tools/lsp/preview/ui.rs

```diff
@@ -212,8 +212,11 @@ pub fn ui_set_known_components(
     api.set_known_components(result.into());
 }
 
-fn to_range(r: (usize, usize)) -> Option<Range> {
-    Some(Range { start: i32::try_from(r.0).ok()?, end: i32::try_from(r.1).ok()? })
+fn to_ui_range(r: TextRange) -> Option<Range> {
+    Some(Range {
+        start: i32::try_from(u32::from(r.start())).ok()?,
+        end: i32::try_from(u32::from(r.start())).ok()?,
```

Comment:
should that not be r.end()
also it seems quite some gymnastics to convert to u32 then i32: is `r.end() as i32`  not good enough?

Reply:
Clippy is happy with this, so the code must be perfect;-)

But yes, that should be an `r.end()`. But `r.start() as i32` gives me a compiler error as `TextSize` is not a primitive type.

`u32::from(r.start()) as i32` works... but then this errors out when the u32 is too big while the cast probably ends up with negative numbers that will make things go up in smoke elsewhere.